src/genie/libs/parser/iosxe/show_cable_tdr_interface.py

```diff
@@ -0,0 +1,154 @@
+        else:
+            output = output
+
+        # initial return dictionary
+        tdr_dict = {}
+
+        # TDR test last run on: January 09 16:19:29
+        #
+        # Interface Speed Local pair Pair length        Remote pair Pair status
+        # --------- ----- ---------- ------------------ ----------- --------------------
+        # Gi1/0/1   auto  Pair A     0    +/- 1  meters N/A         Open
+        #                 Pair B     1    +/- 1  meters N/A         Open
+        #                 Pair C     0    +/- 1  meters N/A         Open
+        #                 Pair D     1    +/- 1  meters N/A         Open
+
+        # TDR test last run on: January 09 16:19:29
+        p1 = re.compile(r'^TDR +test +last +run +on: +(?P<date>\w+ \d+) +(?P<time>\d+:\d+:\d+)$')
+
+        # Gi1/0/1   auto  Pair A     0    +/- 1  meters N/A         Open
+        p2 = re.compile(r'^(?P<interface>\w\w\d\/\d\/\d+)\s+(?P<speed>auto|100M|1000M)'
+                '\s+Pair A     (?P<pair_a_length>\d+)\s+\+\/\- (?P<pair_a_tolerance>\d+)\s+meters '
+                '(?P<pair_a_remote>N\/A|Pair A|Pair B|Pair C|Pair D)\s+(?P<pair_a_status>.*$)')
+
+        #                 Pair B     1    +/- 1  meters N/A         Open
+        p3 = re.compile(r'Pair +B\s+(?P<pair_b_length>\d+)\s+\+\/\- (?P<pair_b_tolerance>\d+)\s+meters'
+                ' +(?P<pair_b_remote>N\/A|Pair A|Pair B|Pair C|Pair D)\s+(?P<pair_b_status>.*$)')
+
+
+        #                 Pair C     0    +/- 1  meters N/A         Open
+        p4 = re.compile(r'Pair +C\s+(?P<pair_c_length>\d+)\s+\+\/\- (?P<pair_c_tolerance>\d+)\s+meters'
+                ' +(?P<pair_c_remote>N\/A|Pair A|Pair B|Pair C|Pair D)\s+(?P<pair_c_status>.*$)')
+
+        #                 Pair D     1    +/- 1  meters N/A         Open
+        p5 = re.compile(r'Pair +D\s+(?P<pair_d_length>\d+)\s+\+\/\- (?P<pair_d_tolerance>\d+)\s+meters'
+                ' +(?P<pair_d_remote>N\/A|Pair A|Pair B|Pair C|Pair D)\s+(?P<pair_d_status>.*$)')
+
+        for line in output.splitlines():
+            line = line.strip()
+
+                
+            # TDR test last run on: January 09 16:19:29
+            m = p1.match(line)
+            if m:
+                date_performed = m.groupdict()['date']
+                time_performed = m.groupdict()['time']
+                continue
+
+            # Gi1/0/1   auto  Pair A     0    +/- 1  meters N/A         Open
+            m = p2.match(line)
+            if m:
+                interface = m.groupdict()['interface']
+                group = m.groupdict()
+                tdr_dict[interface] = {}
+                tdr_dict[interface]['interface'] = m.groupdict()['interface']
+                tdr_dict[interface]['speed'] = m.groupdict()['speed']
+
+                tdr_dict[interface]['date'] = date_performed
+                tdr_dict[interface]['time'] = time_performed
+
+                tdr_dict[interface].setdefault('pairs', {})
+
+                pair = 'pair_a'
```

Comment:
can't you extract the name of the pair using regex? You have p2, p3 and p4 doing the exact same thing

Reply:
@danielgraziano  I didnt want to extract it because the pair name has a space in it (ex 'Pair A').  There are only 4 pairs and thought it simpler just to do each one separately.  I guess I can extract the A B C D for pair and not include the Pair in the pair name.

Reply:
@danielgraziano Made the changes as suggested.  Easier than I thought.
Thanks